### PR TITLE
Move exemplarreservoir to outside of Histogram

### PR DESCRIPTION
**Pull request by cijothomas**
The next PR enables Exemplar for all metrics. This is setting stage by moving ExemplarReservoir logic directly into MetricPoint.

### CODE REVIEW

File: test/OpenTelemetry.Tests.Stress.Metrics/Program.cs [hunk 1]
```diff
@@ -25,15 +25,15 @@ public partial class Program
     private const int ArraySize = 10;
 
     // Note: Uncomment the below line if you want to run Histogram stress test
-    // private const int MaxHistogramMeasurement = 1000;
+    private const int MaxHistogramMeasurement = 1000;
```
Comment (cijothomas) [inline test/OpenTelemetry.Tests.Stress.Metrics/Program.cs:28]: i figured its easy to target Exemplar + Histogram for the stress test now as that is the area being actively worked on now.

File: test/OpenTelemetry.Tests.Stress.Metrics/Program.cs [hunk 2]
```diff
@@ -25,15 +25,15 @@ public partial class Program
     private const int ArraySize = 10;
 
     // Note: Uncomment the below line if you want to run Histogram stress test
```
Comment (utpilla) [inline test/OpenTelemetry.Tests.Stress.Metrics/Program.cs:27]: Remove this text.

File: test/OpenTelemetry.Tests.Stress.Metrics/Program.cs [hunk 3]
```diff
@@ -25,15 +25,15 @@ public partial class Program
...
-    // private const int MaxHistogramMeasurement = 1000;
+    private const int MaxHistogramMeasurement = 1000;
 
     private static readonly Meter TestMeter = new(Utils.GetCurrentMethodName());
     private static readonly Counter<long> TestCounter = TestMeter.CreateCounter<long>("TestCounter");
```
Comment (utpilla) [inline test/OpenTelemetry.Tests.Stress.Metrics/Program.cs:31]: You can put this one in a comment now.

File: test/OpenTelemetry.Tests.Stress.Metrics/Program.cs [hunk 4]
```diff
@@ -44,33 +44,35 @@ public static void Main()
...
-    //    TestHistogram.Record(
-    //        random.Next(MaxHistogramMeasurement),
+    //    TestCounter.Add(
+    //        100,
     //        new("DimName1", DimensionValues[random.Next(0, ArraySize)]),
     //        new("DimName2", DimensionValues[random.Next(0, ArraySize)]),
     //        new("DimName3", DimensionValues[random.Next(0, ArraySize)]));
     // }
+
+    // Note: Uncomment the below lines if you want to run Histogram stress test
```
Comment (utpilla) [inline test/OpenTelemetry.Tests.Stress.Metrics/Program.cs:67]: Remove this line.

File: test/OpenTelemetry.Tests.Stress.Metrics/Program.cs [hunk 5]
```diff
@@ -25,15 +25,15 @@ public partial class Program
...
     // Note: Uncomment the below line if you want to run Histogram stress test
-    // private const int MaxHistogramMeasurement = 1000;
+    private const int MaxHistogramMeasurement = 1000;
 
     private static readonly Meter TestMeter = new(Utils.GetCurrentMethodName());
     private static readonly Counter<long> TestCounter = TestMeter.CreateCounter<long>("TestCounter");
     private static readonly string[] DimensionValues = new string[ArraySize];
     private static readonly ThreadLocal<Random> ThreadLocalRandom = new(() => new Random());
 
     // Note: Uncomment the below line if you want to run Histogram stress test
```
Comment (utpilla) [inline test/OpenTelemetry.Tests.Stress.Metrics/Program.cs:35]: Remove this line.

**Review by utpilla — APPROVED** [review]
Approved with suggestions.

**Pull request merged**